src/Relations/MorphTo.php

```diff
@@ -16,7 +16,11 @@ public function addConstraints()
             // For belongs to relationships, which are essentially the inverse of has one
             // or has many relationships, we need to actually query on the primary key
             // of the related models matching on the foreign key that's on a parent.
-            $this->query->where($this->ownerKey, '=', $this->parent->{$this->foreignKey});
+            $this->query->where(
+                $this->ownerKey,
```

Comment:
Removing the `$this->getKeyName()` fallback in `HybridRelations::morphTo` and allowing MorphTo to be constructed with a null `$ownerKey` would result in an exception here when `null` is used in place of a string field name.

The relevant fix in https://github.com/mongodb/laravel-mongodb/pull/3011 is to fall back to `$this->getForeignKeyName()` here, which respects a custom primary key name on parent.